app/ante/cosmos/setup_test.go

```diff
@@ -0,0 +1,118 @@
+}
+
+func (suite *AnteTestSuite) SetupTest() {
+	checkTx := false
+
+	suite.app = app.EthSetup(checkTx, func(app *app.Evmos, genesis simapp.GenesisState) simapp.GenesisState {
+		if suite.enableFeemarket {
+			// setup feemarketGenesis params
+			feemarketGenesis := feemarkettypes.DefaultGenesisState()
+			feemarketGenesis.Params.EnableHeight = 1
+			feemarketGenesis.Params.NoBaseFee = false
+			// Verify feeMarket genesis
+			err := feemarketGenesis.Validate()
+			suite.Require().NoError(err)
+			genesis[feemarkettypes.ModuleName] = app.AppCodec().MustMarshalJSON(feemarketGenesis)
+		}
+		evmGenesis := evmtypes.DefaultGenesisState()
+		evmGenesis.Params.AllowUnprotectedTxs = false
+		if !suite.enableLondonHF {
+			maxInt := sdkmath.NewInt(math.MaxInt64)
+			evmGenesis.Params.ChainConfig.LondonBlock = &maxInt
+			evmGenesis.Params.ChainConfig.ArrowGlacierBlock = &maxInt
+			evmGenesis.Params.ChainConfig.GrayGlacierBlock = &maxInt
+			evmGenesis.Params.ChainConfig.MergeNetsplitBlock = &maxInt
+			evmGenesis.Params.ChainConfig.ShanghaiBlock = &maxInt
+			evmGenesis.Params.ChainConfig.CancunBlock = &maxInt
+		}
+		if suite.evmParamsOption != nil {
+			suite.evmParamsOption(&evmGenesis.Params)
+		}
+		genesis[evmtypes.ModuleName] = app.AppCodec().MustMarshalJSON(evmGenesis)
+		return genesis
+	})
+
+	suite.ctx = suite.app.BaseApp.NewContext(checkTx, tmproto.Header{Height: 2, ChainID: "evmos_9000-1", Time: time.Now().UTC()})
+	suite.ctx = suite.ctx.WithMinGasPrices(sdk.NewDecCoins(sdk.NewDecCoin(utils.BaseDenom, sdk.OneInt())))
+	suite.ctx = suite.ctx.WithBlockGasMeter(sdk.NewGasMeter(1000000000000000000))
+	suite.app.EvmKeeper.WithChainID(suite.ctx)
+
+	infCtx := suite.ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
+	suite.app.AccountKeeper.SetParams(infCtx, authtypes.DefaultParams())
+
+	encodingConfig := encoding.MakeConfig(app.ModuleBasics)
+	// We're using TestMsg amino encoding in some tests, so register it here.
+	encodingConfig.Amino.RegisterConcrete(&testdata.TestMsg{}, "testdata.TestMsg", nil)
+	eip712.SetEncodingConfig(encodingConfig)
+
+	suite.clientCtx = client.Context{}.WithTxConfig(encodingConfig.TxConfig)
+
+	anteHandler := ante.NewAnteHandler(ante.HandlerOptions{
+		AccountKeeper:          suite.app.AccountKeeper,
+		BankKeeper:             suite.app.BankKeeper,
+		EvmKeeper:              suite.app.EvmKeeper,
+		FeegrantKeeper:         suite.app.FeeGrantKeeper,
+		StakingKeeper:          suite.app.StakingKeeper,
+		IBCKeeper:              suite.app.IBCKeeper,
+		FeeMarketKeeper:        suite.app.FeeMarketKeeper,
+		SignModeHandler:        encodingConfig.TxConfig.SignModeHandler(),
+		SigGasConsumer:         ante.SigVerificationGasConsumer,
+		ExtensionOptionChecker: types.HasDynamicFeeExtensionOption,
+		TxFeeChecker:           evmante.NewDynamicFeeChecker(suite.app.EvmKeeper),
+	})
+
+	suite.anteHandler = anteHandler
+	suite.ethSigner = ethtypes.LatestSignerForChainID(suite.app.EvmKeeper.ChainID())
+}
```

Comment:
This code is repeated on the `/evm/setup_test.go` file and on many others. Do you think we can make only **one** general setup for  the tests and reuse it?

Reply:
yes, that's the goal, this is just moving the tests so that we know what to delete

Reply:
see #1381